server/src/websocket/server.rs

```diff
@@ -896,10 +896,18 @@ where
 
   let data = data.to_string();
   let op2 = op.clone();
+
   let fut = async move {
-    let parsed_data: Data = serde_json::from_str(&data)?;
-    let res = Oper::new(parsed_data).perform(pool, Some(ws_info))?;
-    to_json_string(&op, &res)
+    actix_web::web::block(move || {
+      let parsed_data: Data = serde_json::from_str(&data)?;
+      let res = Oper::new(parsed_data).perform(pool, Some(ws_info))?;
+      to_json_string(&op, &res)
+    })
+    .await
+    .map_err(|e| match e {
+      actix_web::error::BlockingError::Error(e) => e,
+      _ => APIError::err("Operation canceled").into(),
+    })
```

Comment:
@asonix This compiles, but I'm not totally sure if its correct. I'm gonna try to import one of my production DBs and see if it still seems like long DB operations are blocking others.

Reply:
NM, looks like it works!